libafl/src/observers/list.rs

```diff
@@ -0,0 +1,73 @@
+use alloc::{
+    string::{String, ToString},
+    vec::Vec,
+};
+use core::fmt::Debug;
+
+use libafl_bolts::{ownedref::OwnedMutPtr, Error, Named};
+use serde::{Deserialize, Serialize};
+
+use crate::{inputs::UsesInput, observers::Observer};
+
+/// A simple observer with a list of things.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(bound = "T: serde::de::DeserializeOwned")]
+#[allow(clippy::unsafe_derive_deserialize)]
+pub struct ListObserver<T>
+where
+    T: Debug + Serialize,
+{
+    name: String,
+    /// The list
+    list: OwnedMutPtr<Vec<T>>,
+}
+
+impl<T> ListObserver<T>
+where
+    T: Debug + Serialize + serde::de::DeserializeOwned,
+{
+    /// Creates a new [`ListObserver`] with the given name.
+    ///
+    /// # Safety
+    /// Will dereference the list.
+    /// The list may not move in memory.
+    #[must_use]
+    pub unsafe fn new(name: &'static str, list: *mut Vec<T>) -> Self {
```

Comment:
Maybe better to take take an `OwnedMut` thingy directly? IIRC we do that in other places